contracts/DolaFlashMinter.sol

```diff
@@ -9,29 +8,43 @@ import "./utils/Address.sol";
 import "./ERC20/IERC20.sol";
 import "./ERC20/SafeERC20.sol";
 
-contract DolaFlashMinter is Ownable, IERC3156FlashLender {
+/**
+ * @title Dola Flash Minter
+ * @notice Allow users to mint an arbitrary amount of DOLA without collateral
+ *         as long as this amount is repaid within a single transaction.
+ * @dev This contract is abstract, any concrete implementation must have the DOLA
+ *      token address hardcoded in the contract to facilitate code auditing.
+ */
+abstract contract DolaFlashMinter is Ownable, IERC3156FlashLender {
     using SafeERC20 for IERC20;
+    using Address for address;
+    using Address for address payable;
     event FlashLoan(address receiver, address token, uint256 value);
     event FlashLoanRateUpdated(uint256 oldRate, uint256 newRate);
     event TreasuryUpdated(address oldTreasury, address newTreasury);
 
-    IERC20 public constant dola =
-        IERC20(0x865377367054516e17014CcdED1e7d814EDC9ce4);
-    address public treasury = 0x926dF14a23BE491164dCF93f4c468A50ef659D5B;
+    IERC20 public immutable dola;
+    address public treasury;
     uint256 public flashMinted;
     uint256 public flashLoanRate = 0.0008 ether;
```

Comment:
The fee is charged in Dola terms, why aren't we representing the loanRate simply in terms of fraction or percentage? Why is ether used here (and I see later divided by 1e18)?

Reply:
Not sure I follow, this currently represents a 0.08% fee.

Reply:
My question basically is why have we initialised flashLoanRate as `0.0008 ether` rather than `0.008`?
Why should a call to get FlashLoanRate return 0.0008*10^18, rather than simply 0.008?

The answer might be something obvious, I'm not sure what it is though.

Reply:
Ethereum doesn't support floating point numbers.

Reply:
Yeah thanks, I missed that. Had a chat offline with TSnark and he cleared that up for me.